real_time/arduino_src/lib_avr/servoreceiver/servoreceiver.cpp

```diff
@@ -141,19 +148,43 @@ void ServoReceiver::OnInterruptReceiver(){
 /*
 no arguments
 Returns the timestamp of the last successfully received servo pulse in
-microseconds since startup or last overflow.
+microseconds since startup or last overflow.  This operation is atomic
 */
-unsigned long ServoReceiver::GetLastTimestamp() {
-    return last_timestamp_;
+unsigned long ServoReceiver::GetLastTimestamp() 
+{
+    uint8_t oldSREG;
+    unsigned long temp;
+
+    oldSREG = SREG;
+
+    cli();
```

Comment:
don't you need to call sei() at some point to reenable interrupts? You deleted that line in main